apps/src/lib/script-editor/editorRedux.js

```diff
@@ -160,50 +160,50 @@ function updateLevelPositions(levels) {
 
 export const NEW_LEVEL_ID = -1;
 
-function stages(state = [], action) {
+function lessons(state = [], action) {
   let newState = _.cloneDeep(state);
 
   switch (action.type) {
     case INIT:
-      return action.stages;
+      return action.lessons;
     case REORDER_LEVEL: {
-      const levels = newState[action.stage - 1].levels;
+      const levels = newState[action.lesson - 1].levels;
```

Comment:
Where is the action object that's passed in defined? I'm not seeing a corresponding change to its keys (sorry if I missed it.)

For renames going forward I think it'd be good for us to explicitly call out which "interface" changes we're making in the PR, since those are hardest to get right I think. By "interface" change I mean fields in an object or keys in a ruby hash that gets passed in or return from a function. And then if you've already taken the care to cover that the reviewer can start already knowing that. (I'll try to do this too.)

Reply:
Can you give examples of what you mean @uponthesun ?

Reply:
In this case, since you're changing this code to call `action.lesson` instead of `action.stage`, it also requires a change to whatever object is being passed to use `lesson` as the field name. So it'd help to say something like this in the description:

"The `lesson` function is called in XYZ places, and the field names for the action objects being passed are defined in ABC places, which have been updated."

Or, a different approach could be to make a commit which only contains all the changes needed for renaming that field. I tried to do that in https://github.com/code-dot-org/code-dot-org/pull/34651, for example this commit: https://github.com/code-dot-org/code-dot-org/pull/34651/commits/b21cdf647e0935ca33aa7b47a1f539da89c11c75

Reply:
I'm just more paranoid now after attempting to rename too much in one go during the initial Lesson > Stage 😬

Reply:
Ok that all makes sense. I will try to do that more clearly in the next PR.